### DISCUSSION TITLE
Possible bug in icon lookup

### DISCUSSION THREAD

**Discussion by ghost** [davatorium/rofi#1719] [(Possible) Issues] [answered]
Running `rofi -show drun -show-icons` from a terminal and going over the list with the keyboard, I am getting the following message exactly once:

`(process:1772): Helpers.IconFetcher-WARNING **: 15:42:37.403: Failed to load image: Failed to open file “/usr/share/icons/gnome/32x32/apps/terminal.png”: No such file or directory`

This seems a bit weird in two different ways:

1) I have specified an icon theme and a fallback icon in `config.rasi` like this:
```
configuration {
    modes:               "drun,window,filebrowser";
    font:                "Cantarell 14";
    icon-theme:          "Papirus-Dark";
    drun-display-format: "{name}";
    drun-match-fields:   "name";
    drun {
        fallback-icon: "/usr/share/icons/Papirus-Dark/32x32/apps/application-default-icon.svg";
    }
}
```
and the fallback icon works, and I have also set the same icon theme in several other places in my system (dconf, xsettingsd.conf, gtk settings.ini, but NOT in .Xresources).

2) It seems to be printed when I hit the first result in the third page (29th line) even if the entry is different. In the beginning this is VSCode, but after launching VSCode a few times so that it goes higher in the list, the message is not printed on VSCode but on Viber which now is the first result of the third page.

-----

This happens in a recent build of `next` as well as in `1.7.5` from the Arch repository.

**Comment by DaveDavenport** [comment, marked as answer]
2.) Icons are fetched async. So when it is printed does not correlate fully with what is visible at the point in time it is printed.  If you scroll through the list, the lookups are queued (multi-threaded) and fetched, once and icon is available the interface is refreshed to show it.
This makes it you can get the error even though that element is no longer visible.

1.)  Can you check if '/usr/share/icons/gnome/32x32/apps/terminal.png'  exists? or is a broken symlink? or a wrong hardcoded path in a .desktop file?    Either the icon theme lookup tells rofi an icon should exists there or a desktop file.

I also notice I get some weird icon paths (firefox seems to look in /opt/firefox that does not exists for me)  back from the icon system. @sardemff7  is there an 'icon' cache file that is used? 

edit: ^^ that was a broken .desktop file for me.

**Reply by ghost** [reply]
I do indeed have a .desktop file from Vivado with the line `Icon=/usr/share/icons/gnome/32x32/apps/terminal.png` and this path does not exist on my system. I should have looked before posting...

**Reply by DaveDavenport** [reply]
Vivado is so much better then the old ISE, but still a horrible beast :D.
